Automattic/wp-calypso
# Settings: Remove legacy PublicPostTypes component

We are supposed to support only Jetpack current version - 1, so we can clean up any older version functionality from Calypso. The Public Post Types sync toggle is unnecessary since Jetpack 4.2 (which was released over a year and a half ago). This PR removes the related component, its usage and the related feature flag.

Part of #29888

#### Changes proposed in this Pull Request

* Remove public post types sync feature flag from config
* Remove `<PublicPostTypes />` from `<DataSynchronization />`
* Delete `<PublicPostTypes />` component.

#### Testing instructions

* Checkout this branch, or spin it up on calypso.live.
* Pick a Jetpack site (with Jetpack >= 6.7.0).
* Go to Settings > General > Manage Connection.
* Make sure the "Data Synchronization" card still works like it did before.
